src/query/xasl_cache.c

```diff
@@ -1666,10 +1669,69 @@ xcache_insert (THREAD_ENTRY * thread_p, const compile_context * context, XASL_ST
   return error_code;
 }
 
+/*
+ * xcache_invalidate_qcaches () - Invalidate all query cache entries which pass the invalidation check.
+ *				  all invalidated query cache entries are removed.
+ *
+ * return		 : Void.
+ * thread_p (in)	 : Thread entry.
+ * invalidate_check (in) : Invalidation check function.
+ * arg (in)		 : Argument for invalidation check function.
+ */
+int
+xcache_invalidate_qcaches (THREAD_ENTRY * thread_p, const OID * oid)
+{
+  int res = NO_ERROR;
+  bool finished = false;
+  XASL_CACHE_ENTRY *xcache_entry = NULL;
+
+  if (!xcache_Enabled)
+    {
+      return NO_ERROR;
+    }
+
+  xcache_hashmap_iterator iter = { thread_p, xcache_Hashmap };
+
+  while (!finished)
+    {
+      /* make sure to start from beginning */
+      iter.restart ();
+
+      /* Iterate through hash, check entry OID's and if one matches the argument, mark the entry for delete and save
+       * it in delete_xids buffer. We cannot delete them from hash while iterating, because the one lock-free
+       * transaction can be used for one hash entry only.
+       */
+      while (true)
+	{
+	  xcache_entry = iter.iterate ();
```

Comment:
When DML is in progress, oid is being searched in all cache entries.
Execution speed of DML will be slow down.
It will have to be tested to see if this slowdown is acceptable.